### PR TITLE
Use GitHub yaml templates for ICE, Docs and Diagnostics tickets

### PR DESCRIPTION
**Pull request by estebank**
The GitHub yaml templates allow us to define HTML forms with validation for issue templates, instead of the current markdown based templates which only let us introduce text into the user editable text area. The form lets us make some fields mandatory, as well as add text that won't pollute the user's text and titles that won't be interfered with by enterprising users.

### CODE REVIEW

**Comment by estebank** [comment]
You can see what this will look like by navigating from https://github.com/estebank/rust-issue-template-test/issues/new/choose

**Comment by Mark-Simulacrum** [comment]
Not opposed to trying this! I know required form inputs can be annoying when filing issues though, but we already have alternatives and users can always opt-out, so it's probably OK.

@bors r+ rollup

File: .github/ISSUE_TEMPLATE/ice.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        If possible, try to provide a minimal verifiable example.
+
+        You can read "[Rust Bug Minimization Patterns](http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/)" for how to create smaller examples.
+
+  - type: textarea
+    id: code
+    attributes:
+      label: Code
+      description: Please provide code or a link to a repository that can reproduce the problem
```
Comment (mejrs) [inline .github/ISSUE_TEMPLATE/ice.yaml:19]: Can 
```suggestion
      description: Please provide code, [playground[(https://play.rust-lang.org/) link or a link to a repository that can reproduce the problem
```

I'd like to nudge people to reproduce on the playground :)

**Review by mejrs — COMMENTED** [review]
I generally prefer the placeholders in these fields to be more suggestive, do you mind if I send a PR myself? 🙂

File: .github/ISSUE_TEMPLATE/ice.yaml (hunk 1 above)
Comment (estebank) [reply .github/ISSUE_TEMPLATE/ice.yaml:19]: Yes! Please send a follow up PR!

**Comment by albertlarsan68** [comment]
Hello,
On GitHub Mobile, the new categories are down the list and need to be filled in a browser.
I think this could be reverted until GitHub Mobile supports them natively.

**Comment by estebank** [comment]
@albertlarsan68 Does GitHub Mobile not open an embedded browser for them to work? If so, that's quite a bad user experience :-/

This should be a consideration, but I am hesitant to revert the changes for ICEs in particular because I'm failing to picture a situation where someone encounters one in a device where the user is also needing to file the report using GitHub Mobile.

**Comment by albertlarsan68** [comment]
It seems like it is one of the few cases where it doesn't work, while for other links it opens them in an embedded browser.

**Comment by estebank** [comment]
@albertlarsan68 does it at least give a reasonable error? I want to figure out how quickly we need to come to a decision for changes.

**Comment by albertlarsan68** [comment]
It is not an error, just that it opens in the "external" browser, and not an embedded browser.

**Pull request merged**